app/src/main/java/cloudcity/networking/models/NetworkDataModel.java

```diff
@@ -28,9 +28,12 @@ public boolean hasNonZeroLatitudeAndLongitude() {
     }
 
     private boolean isNonZero(double value) {
+        boolean isNotZero = value != 0f;
+        boolean isNotExactZero = value != 0.0;
```

Comment:
well to be fair, all of them except the epsilon one are insignificant and unnecessary, but then again, removing them will shave insignificant amount of execution time.
they are kept just because a freshly initialized bogus Location will match them.